dbt_platform_helper/domain/pipelines.py

```diff
@@ -63,147 +72,67 @@ def generate(self, terraform_platform_modules_version, deploy_branch):
 
         self._clean_pipeline_config(copilot_pipelines_dir)
 
+        terraform_platform_modules_version = get_required_terraform_platform_modules_version(
+            cli_terraform_platform_modules_version,
+            platform_config_terraform_modules_default_version,
+        )
+
         if has_environment_pipelines:
-            environment_pipelines = pipeline_config[ENVIRONMENT_PIPELINES_KEY]
+            environment_pipelines = platform_config[ENVIRONMENT_PIPELINES_KEY]
+            accounts = {
+                config.get("account")
+                for config in environment_pipelines.values()
+                if "account" in config
+            }
 
-            for config in environment_pipelines.values():
-                aws_account = config.get("account")
+            for account in accounts:
                 self._generate_terraform_environment_pipeline_manifest(
-                    pipeline_config["application"],
-                    aws_account,
+                    platform_config["application"],
+                    account,
                     terraform_platform_modules_version,
-                    platform_config_terraform_modules_default_version,
                     deploy_branch,
                 )
 
         if has_codebase_pipelines:
-            account_id, _ = get_account_details()
-
-            for codebase in pipeline_config[CODEBASE_PIPELINES_KEY]:
-                self._generate_codebase_pipeline(
-                    account_id,
-                    app_name,
-                    codestar_connection_arn,
-                    git_repo,
-                    codebase,
-                    base_path,
-                    copilot_pipelines_dir,
-                    templates,
-                )
+            codebase_pipelines = platform_config[CODEBASE_PIPELINES_KEY]
+            provisioned_ecrs = set(self.ecr_provider.get_ecr_repo_names())
+            required_ecrs = {
+                codebase: f"{platform_config['application']}/{codebase}"
+                for codebase in codebase_pipelines.keys()
+                if f"{platform_config['application']}/{codebase}" in provisioned_ecrs
+            }
+            required_imports = {
```

Comment:
So it might be the time of the day but I am stuck reading what exactly this is doing here. 
Is this correct?
- you build a dictionary of `required_ecrs` if the ecr repo is in `provisioned_ecrs`. You build with the key: value of the codebase and the repo.
- you then build a dictionary of `required_imports` if the ecr repo is in `provisioned_ecrs`.  You build with the key: value of the `codebase` and the `repo`.
will the values of `required_ecrs` and `required_imports` ever be different?

Reply:
Yes, they'll be different if there is a new environment. i.e. if there is a required ecr that hasn't already been created. The required imports are the ECRs that are required but already exist (so need to be imported).

I'll check to see if there's a test that describes this case and add one if not. And perhaps rename some variables to make them more descriptive.

Reply:
The updated code now makes a lot more sense